audiobook_generator/tts_providers/openai_tts_provider.py

```diff
@@ -16,15 +16,33 @@
 def get_supported_formats():
     return ["mp3", "aac", "flac", "opus", "wav"]
 
+def get_supported_voices():
+    return ["alloy", "ash", "ballad", "coral", "echo", "fable", "onyx", "nova", "sage", "shimmer", "verse"]
+
+def get_supported_models():
+    return ["tts-1", "tts-1-hd", "gpt-4o-mini-tts"]
+
+def get_price(model):
+    # https://platform.openai.com/docs/pricing#transcription-and-speech-generation
+    if model == "tts-1": # $15 per 1 mil chars
+        return 0.015
+    elif model == "tts-1-hd": # $30 per 1 mil chars
+        return 0.03
+    elif model == "gpt-4o-mini-tts": # $12 per 1 mil tokens (not chars, as 1 token is ~4 chars)
+        return 0.003
+    else:
```

Comment:
We could replace the `ValueError` here with a simple `logger.warn` to indicate that the price calculation is not applicable for the user input model. Otherwise this is still an overly strict constraint, and the program will stop working.